### ISSUE TITLE
Add stress tests for settings table

### ISSUE THREAD

**Issue by AlekSi** [FerretDB/FerretDB#1282]
* [x] Add a stress integration test that creates X collections in parallel in X goroutines.
* [ ] As the test will always fail and implementation is not in scope, either skip the test or not merge it.

**Comment by rumyantseva** [comment]
`[upd]` the problem with collection creation is an easy one - when we query `CreateDatabaseIfNotExists` from `msg_create` we don't consider a case when DB already exists as normal. But we should consider it normal as it's ok if the DB is created from one of the parallel requests. The fix is in #1316.

**Comment by rumyantseva** [comment]
I also fixed simple test failures occurred on my local env by adding transactions to `SchemaStats` usage.
